core/node/da_clients/src/eigen/verifier.rs

```diff
@@ -48,20 +42,37 @@ impl VerifierClient for PKSigningClient {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum VerificationError {
+    #[error("Service manager error")]
     ServiceManagerError,
-    KzgError,
+    #[error(transparent)]
+    KzgError(#[from] rust_kzg_bn254::errors::KzgError),
+    #[error("Wrong proof")]
     WrongProof,
-    DifferentCommitments,
-    DifferentRoots,
+    #[error("Different commitments: expected {expected:?}, got {actual:?}")]
+    DifferentCommitments {
+        expected: G1Affine,
+        actual: G1Affine,
+    },
+    #[error("Different roots: expected {expected:?}, got {actual:?}")]
+    DifferentRoots { expected: String, actual: String },
+    #[error("Empty hash")]
     EmptyHash,
-    DifferentHashes,
+    #[error("Different hashes: expected {expected:?}, got {actual:?}")]
+    DifferentHashes { expected: String, actual: String },
+    #[error("Wrong quorum params")]
     WrongQuorumParams,
+    #[error("Quorum not confirmed")]
     QuorumNotConfirmed,
-    CommitmentNotOnCurve,
-    CommitmentNotOnCorrectSubgroup,
-    LinkError,
+    #[error("Commitment not on curve: {0}")]
+    CommitmentNotOnCurve(G1Affine),
+    #[error("Commitment not on correct subgroup: {0}")]
+    CommitmentNotOnCorrectSubgroup(G1Affine),
+    #[error("Link Error: {0}")]
+    LinkError(String),
+    #[error(transparent)]
+    JoinError(#[from] tokio::task::JoinError),
```

Comment:
I think we should map the join error to a different and existing error variant instead of adding it to the VerificationError